common-rendering/src/components/accordion.tsx

```diff
@@ -0,0 +1,121 @@
+// ----- Imports ----- //
+
+import { css } from "@emotion/react";
+import type { SerializedStyles } from "@emotion/react";
+import { neutral, line, background } from "@guardian/src-foundations/palette";
+import { headline } from "@guardian/src-foundations/typography";
+import { remSpace } from "@guardian/src-foundations";
+import { focusHalo } from "@guardian/src-foundations/accessibility";
+import { SvgChevronUpSingle, SvgChevronDownSingle } from "@guardian/src-icons";
+import { from } from "@guardian/src-foundations/mq";
+import { darkModeCss } from "../lib";
+
+// ----- Component ----- //
+
+interface AccordionProps {
+	children: React.ReactNode;
+	supportsDarkMode: boolean;
+	accordionTitle: string;
+	backgroundBody: 'white' | 'grey';
+}
+
+const detailsStyles: SerializedStyles = css`
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const titleRowStyles = (supportsDarkMode: boolean): SerializedStyles => css`
+	position: relative;
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	border-top: ${line.primary} 1px solid;
+	background-color: ${background.primary};
+	padding: ${remSpace[1]} ${remSpace[2]} 0 ${remSpace[3]};
+	&:focus {
+		${focusHalo};
+	}
+	path {
+		fill: ${neutral[46]};
+	}
+	svg {
+		height: 2rem;
+	}
+	${darkModeCss(supportsDarkMode)`
+		path {
+			fill: ${neutral[60]};
+		}
+	`}
+	${from.phablet} {
+		padding: ${remSpace[1]} ${remSpace[4]} 0 ${remSpace[5]};
+	}
+	${from.desktop} {
+		display: none;
+	}
+`;
+
+const titleStyle = (supportsDarkMode: boolean): SerializedStyles => css`
+	${headline.xxsmall({ fontWeight: "bold", lineHeight: "tight" })};
+	color: ${neutral[7]};
+	${darkModeCss(supportsDarkMode)`
+		color: ${neutral[86]};
+	`}
+`;
+
+const backgroundColour = (backgroundBody: 'white' | 'grey'): SerializedStyles => {
```

Comment:
Does the background colour change in dark mode?

Reply:
What's the plan for colours? I feel it would be great to have a co-ordinated plan for this kind of thing rather than allowing different approaches to spread

Reply:
Good question - I'm guessing this would need to work differently for dark mode.

Reply:
The way I see it, if a component takes `supportsDarkMode` then that component already knows what it should look like in dark mode. It just includes the necessary styles if that's set to `true` (AR), and doesn't include them if it's set to `false` (DCR, Editions).

You can see this pattern further up this file in the title styles.

Reply:
Off topic for this PR but I haven't really got my head around how `supportsDarkMode` works (which is not to say it will ot won't!). Do you think it will play well DCR if we want to use css variables to implement dark mode?

Reply:
Also, this code (and I know it's just copied over but it's a good trigger for this discussion) is setting colours directly, not using the palette pattern. Is that intentional or is the expectation that we will migrate such usages to use `decidePalette` at some point?

Reply:
@marjisound and I have just done some work to make sure that dark mode is working correctly for the accordion.

Reply:
@oliverlloyd `darkModeCss` implementation in common is that the function only returns dark mode `SerializedStyles` css if `supportsDarkMode` is set to `true` and returns css`` if `supportsDarkMode` is set to `false`. Are you thinking of using css variables in DCR and not make use of the `darkModeCss` function?

Reply:
@oliverlloyd  we're using ` (backgroundBody: 'white' | 'grey')` since in the same liveblog page, we have 2 accordion that need different  background-color (keyevent & live block). This is also a similar issue I was chatting with you today for ShareIcons. Do you think the approach Lucy has taken could be improved?

Reply:
> Are you thinking of using css variables in DCR and not make use of the `darkModeCss` function?

Basically, yes. I think for DCR, to avoid having to rerender the page, we would want to use css-variables. The bit where I'm unsure about is if this approach could live alongside this one?

> we have 2 accordion that need different background-color (keyevent & live block)

Yes, this sounds very similar to the issue we discussion earlier. There's a component that needs to be rendered in multiple contexts and it behaves differently in each. We can give it `format` and use a media query but there's still a missing piece of information to make the choice about which colour to use (i.e. the context for where the component is, like in KeyEvents). So, yeah, you can either:

- Pass in this context information as a prop - `isInKeyevnets`
- Or pass in the colour itself as a prop

I don't think there's a right or wrong. It's about trade offs